packages/modern-test-app/app/templates/helpers.hbs

```diff
@@ -12,6 +12,10 @@
 {{xor true false}}
 
 <AndOrTypeChecking
-  @andArg={{or (hash) (array)}}
-  @orArg={{and (hash) (array)}}
+  @andArg={{and (hash) (array)}}
+  @orArg={{or (hash) (array)}}
+  {{!-- and returns the last truthy item --}}
+  @andFallbackArg={{and true "string"}}
+  {{!-- or returns the first truthy item --}}
+  @orFallbackArg={{or undefined "string"}}
```

Comment:
Both of these args only accept strings.

But the types errors indicate that the and/or helpers are trying to provide them
`boolean | string`
`undefined | string`

Reply:
It might seem that these are pedantic or not realistic, but I'm trying to cover a scenario like;
```ts
teacherId: string | undefined // may or may not be set
AllTeachersOptionId: "all-teachers" // always set
```
template
`@teacherId={{or this.teacherId AllTeachersOptionId}}`